lib/csu/aarch64/crt0.S

```diff
@@ -8,8 +8,18 @@
 # to prepare_program_stack in exec.c.
 
 ENTRY(_start)
+        # Grab argc from stack.
+        ldr     w0, [sp, #0]
+
+        # Prepare argv.
+        add     x1, sp, #8
+
+        # Prepare envp.
+        add     x2, sp, #16
```

Comment:
`x2` points to `argv[1]`

Reply:
I'm confused. In `exec_args_copyout` it is said that layout is sp, argc, argv, envp, argv[0..n], envp[0..m] so first value above argv – which is x2 – should be envp. What did I miss?

Reply:
Hmm... could you check if _MIPS_ implementation is correct then?

Reply:
It looks like _MIPS_ implementation works. It makes sense for me because we still calculate correct address according to `exec_args_copyout` comments but in different way.